include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -83,6 +83,50 @@ void tess_ctrl_barrier() {
 void memoryBarrierShared() {
     spirv::memoryBarrier(spv::ScopeDevice, spv::MemorySemanticsAcquireReleaseMask | spv::MemorySemanticsWorkgroupMemoryMask);
 }
+
+
+namespace impl 
+{
+
+template<typename T, bool isSigned, bool isIntegral>
+struct bitfieldExtract {};
+
+template<typename T, bool isSigned>
+struct bitfieldExtract<T, isSigned, false>
+{
+    T operator()( T val, uint32_t offsetBits, uint32_t numBits )
+    {
+        static_assert(is_integral<T>::value, "T is not an integral type!" )
+        return val;
+    }
+};
+
+template<typename T>
+struct bitfieldExtract<T, true, true>
+{
+    T operator()( T val, uint32_t offsetBits, uint32_t numBits )
+    {
+        return spirv::bitFieldSExtract<T>( val, offsetBits, numBits );
+    }
+};
+
+template<typename T>
+struct bitfieldExtract<T, false, true>
+{
+    T operator()( T val, uint32_t offsetBits, uint32_t numBits )
+    {
+        return spirv::bitFieldUExtract<T>( val, offsetBits, numBits );
+    } 
+};
+
+}
+
+template<typename T>
+T bitfieldExtract( T val, uint32_t offsetBits, uint32_t numBits )
+{
+    return impl::bitfieldExtract<T, is_signed<T>::value, is_integral<T>::value>( val, offsetBits, numBits );
```

Comment:
did you actually test your code though, cause I can see with my very own eyes this is invalid syntax.